MixItUp.Base/Actions/OverlayAction.cs

```diff
@@ -370,8 +378,22 @@ public T Copy<T>()
             return jobj.ToObject<T>();
         }
 
-        private string GetAnimationClassName(string name)
+        private string GetAnimationClassName<T>(T typedName)
         {
+            string name = typedName.ToString();
+            if (name == "Random")
+            {
+                List<T> values = new List<T>();
+                foreach(T value in Enum.GetValues(typeof(T)))
+                {
+                    if (value.ToString() != "None" && value.ToString() != "Random")
```

Comment:
Since you're looping through the whole array to remove these two items, why not just do something like the following:

            List<T> values = EnumHelper.GetEnumList<T>().ToList();
            values.RemoveAll(v => v.ToString().Equals("None") || v.ToString().Equals("Random"));
            name = values[Random.Next(values.Count)].ToString();

Accomplishes the same thing, just a bit more compacter code :)